src/Common.Lib/EventLoggingBase.cpp

```diff
@@ -155,16 +138,8 @@ HRESULT CEventLoggingBase::TerminateEventSource()
         }
     }
 
-    // Allow thread to drain the queue. Normally, we'd just wait for the thread
-    // to finish (wait on the thread) but the thread is deadlocking on CoUninitialize.
-    // As a workaround, use a seperate event that the thread will signal when it is
-    // finished processing items and no longer using the event source.
-    // Only wait on the handle if the thread was created. This avoids situations where
-    // event logging was not initialized but TerminateEventSource was still called.
-    if (m_hEventQueueThread)
```

Comment:
I would still check that the thread exists before waiting on the thread just in case the thread wasn't started or failed to start.